pkg/controller/catalog/storage/mem_storage.go

```diff
@@ -16,211 +16,45 @@ limitations under the License.
 
 package storage
 
-import (
-	"fmt"
-	"log"
-	"strings"
-
-	model "github.com/kubernetes-incubator/service-catalog/model/service_controller"
-)
-
-type bindingPair struct {
-	binding    *model.ServiceBinding
-	credential *model.Credential
-}
-
 type memStorage struct {
-	brokers map[string]*model.ServiceBroker
-	// This gets fetched when a SB is created (or possibly later when refetched).
-	// It's static for now to keep compatibility, seems like this could be more dynamic.
-	catalogs map[string]*model.Catalog
-	// maps instance ID to instance
-	services map[string]*model.ServiceInstance
-	// maps binding ID to binding
-	// TODO: support looking up all bindings for a service instance.
-	bindings map[string]*bindingPair
+	brokers        *memStorageBroker
```

Comment:
#199 is completely in line with the k8s precedent, but there's not an exact match between their current usage in this project and what we will eventually settle on.  The analog is that for the REST storage for a type, there is a package for each resource.  However, the controllers in k8s do not use that API directly.  Instead, they use the golang client for the k8s API.  So, long-term, our controller is going to use the golang client for the service-catalog API and k8s controller infrastructure to maintain watch-populated caches in memory. 

Does that make sense?